retworkx-core/src/shortest_path/dijkstra.rs

```diff
@@ -20,14 +20,54 @@
 use std::collections::BinaryHeap;
 use std::hash::Hash;
 
-use indexmap::map::Entry::{Occupied, Vacant};
-
 use petgraph::algo::Measure;
-use petgraph::visit::{EdgeRef, IntoEdges, VisitMap, Visitable};
+use petgraph::graph::IndexType;
+use petgraph::visit::{EdgeRef, IntoEdges, NodeIndexable, VisitMap, Visitable};
 
 use crate::dictmap::*;
 use crate::min_scored::MinScored;
 
+// Trait for supporting storing
+pub trait DistanceMap<K, V> {
+    fn new_distancemap(num_elements: usize) -> Self;
```

Comment:
I'd prefer naming this method like
```suggestion
    fn distance_map(num_elements: usize) -> Self;
```
or just `new` or `build`